### ISSUE TITLE
Audio issues with `pipewire-qubes` and `flatpak` apps

### ISSUE THREAD

**Issue by SaswatPadhi** [QubesOS/qubes-issues#8896]
### Qubes OS release

R4.2

### Brief summary

Audio in AppVMs with `pipewire-qubes` + `flatpak` apps seems to break sometimes.
Sometimes it works, but most of the time it doesn't even after multiple restarts (of the AppVM).

I finally switched back to `pulseaudio-qubes`, and the audio works fine now.
However, for my `flatpak` apps I have to install `xdg-desktop-portal-gtk`, which installs `pipewire` as a "weak dependency. Although that doesn't affect Pulseaudio, it would be nice to have pipewire and flatpak working together nicely.

### Steps to reproduce

1. Install `pipewire-qubes`, `flatpak` and `xdg-desktop-portal-gtk` in a fedora-39-minimal TemplateVM.
2. Create an AppVM based on the TemplateVM
3. Install a `flatpak --user` app in the AppVM that can play audio (e.g., firefox)
4. Play audio in the flatpak app, keep your fingers crossed

### Expected behavior

Audio works in the flatpak app

### Actual behavior

Audio doesn't, a lot of times.

---

I have been closely following the discussion on #8805 and #8816.

Of the two solutions that @marmarek mentioned:

> As for VM, again, both options should work separately, but you shouldn't have them at the same time. So, either:

> - pulseaudio + pulseaudio-qubes (and systemctl --user status pulseaudio should be running), or
> - pipewire + pipewire-pulse + pipewire-qubes (and systemctl --user status pipewire-pulse.socket should be running)

The second option works for me on both debian-12-minimal and fedora-{38,39}-minimal.
However, when I have flatpak installed as well, audio from flatpak apps seems broken. Audio works again if I switch to the first option.

**Comment by marmarek** [comment]
For using pipewire, make sure you have pulseaudio removed (or at least not running). And also make sure you have `pipewire-pulse` installed and running (`systemctl --user` is your friend).

**Comment by SaswatPadhi** [comment]
Yes, on my minimal templates I only installed `pipewire-qubes` and that works fine. `pulseaudio-qubes` and `pulseaudio` are not installed.

With `flatpak` installed however, that setup doesn't work well. Instead I had to remove all `pipewire` packages and install `pulseaudio-qubes` etc. Then the audio works as expected. And the audio still works when I install `xdg-desktop-portal-gtk` which installs `pipewire` (but not `pipewire-qubes`) as a weak dependency.

**Comment by marmarek** [comment]
What is an example flatpak app that you see the issue most?

**Comment by SaswatPadhi** [comment]
I have Firefox and Spotify and audio wasn't working on either.

**Comment by DemiMarie** [comment]
> I have Firefox and Spotify and audio wasn't working on either.

That’s certainly a bug.  Can you check that `pipewire-pulse.service` is running as a `--user` service?

**Comment by TNTBOMBOM** [comment]
Just tested this now with fedora-39-minimal and debian-12-minimal, it worked for me: 

- in fedora-39-minimal (preferable to be fresh) install: `pipewire qubes-pipewire` 

Then as a user (Dont use sudo/root) run: `systemctl --user enable pipewire-pulse.service`

** Then output of status **without** root should give: 

```
[user@fedora-39-minimal ~]$ systemctl --user status pipewire-pulse.service
● pipewire-pulse.service - PipeWire PulseAudio
     Loaded: loaded (/usr/lib/systemd/user/pipewire-pulse.service; enabled
; preset: disabled)
    Drop-In: /usr/lib/systemd/user/service.d
             └─10-timeout-abort.conf
     Active: active (running) since Fri 2024-02-23 00:01:04 GMT; 3min
 44s ago
TriggeredBy: ● pipewire-pulse.socket
   Main PID: 583 (pipewire-pulse)
      Tasks: 3 (limit: 374)
     Memory: 2.4M
        CPU: 14ms
     CGroup: /user.slice/user-1000.slice/user@1000.service/session.slice/pipewir
e-pulse.service
             └─583 /usr/bin/pipewire-pulse

Feb 23 00:01:04 fedora-39-minimal systemd[571]: Started pipewire-pulse.service -
 PipeWire PulseAudio.
[user@fedora-39-minimal ~]$ 
```

- in debian-12-minimal (preferable to be fresh) install: `pipewire pipewire-pulse pipewire-qubes`

Then as a user (**Dont** use sudo/root) run: 

`systemctl --user daemon-reload`

`systemctl --user --now enable pipewire pipewire-pulse`

** Then output of status **without** root should give: 

```
user@debian-12-minimal:~$ systemctl --user status pipewire-pulse.service
● pipewire-pulse.service - PipeWire PulseAudio
     Loaded: loaded (/usr/lib/systemd/user/pipewire-pulse.service; enabled; pre>
     Active: active (running) since Fri 2024-02-23 00:00:58 GMT; 13min ago
TriggeredBy: ● pipewire-pulse.socket
   Main PID: 589 (pipewire-pulse)
      Tasks: 2 (limit: 374)
     Memory: 2.6M
        CPU: 15ms
     CGroup: /user.slice/user-1000.slice/user@1000.service/session.slice/pipewi>
             └─589 /usr/bin/pipewire-pulse

Feb 23 00:00:58 debian-12-minimal systemd[569]: Started pipewire-pulse.service >
Feb 23 00:00:58 debian-12-minimal pipewire-pulse[589]: mod.rt: Can't find org.f>
Feb 23 00:00:58 debian-12-minimal pipewire-pulse[589]: mod.rt: found session bu>
lines 1-14/14 (END)...skipping...
● pipewire-pulse.service - PipeWire PulseAudio
     Loaded: loaded (/usr/lib/systemd/user/pipewire-pulse.service; enabled; preset: enabled)
     Active: active (running) since Fri 2024-02-23 00:00:58 GMT; 13min ago
TriggeredBy: ● pipewire-pulse.socket
   Main PID: 589 (pipewire-pulse)
      Tasks: 2 (limit: 374)
     Memory: 2.6M
        CPU: 15ms
     CGroup: /user.slice/user-1000.slice/user@1000.service/session.slice/pipewire-pulse.service
             └─589 /usr/bin/pipewire-pulse

Feb 23 00:00:58 debian-12-minimal systemd[569]: Started pipewire-pulse.service - PipeWire PulseAudio.
Feb 23 00:00:58 debian-12-minimal pipewire-pulse[589]: mod.rt: Can't find org.freedesktop.portal.Desktop. Is xdg->
Feb 23 00:00:58 debian-12-minimal pipewire-pulse[589]: mod.rt: found session bus but no portal
~
```

Now change your dvm based dispvms (sys-net,-firewall,-usb..) to your preferable template (im using fedora-39-minimal), and accordingly so as your dom0 (from qubes global config). 

worked as expected in debianVM based on debian-12-minimal for FF and Chrome from flatpak, signal from its own repo, and so as the microphone.


---------------

Note: instructions above outlined how to make sound work through pipewire in minimal templates but not how to make the templates fully functional for everything you need e.g i didnt say which package to install for your usb to work or your dom0 update to function..etc.
